### PR TITLE
Fix _ez_content_view route controller and allow fallback to legacy /content/view/full/42

### PR DESCRIPTION
**Pull request by emodric**
This PR fixes couple of issues with the new `_ez_content_view` route:

1) `ezpublish.controller.content` service does not exist, `ez_content` must be used instead. Additionally, when using the service notation here, `Action` should not be appended to action name. Thus, correct controller reference is `ez_content:viewContent`

2) `/content/view/{contentId}/{language}` route path clashes with legacy `/content/view/full/42` URLs. In order to avoid that, route path was renamed to `/view/content/{contentId}/{language}`.

3) The route was missing the `viewType` and `layout` parameters, which means that something like `/view/content/42/eng-GB` was failing with `Controller "eZ\Publish\Core\MVC\Symfony\Controller\Content\ViewController::viewContent()" requires that you provide a value for the "$viewType" argument (because there is no default value or because there is a non optional argument after this one).` error.

4) Now that route has two additional parameters, `contentId` and `language`, they have to be removed in URL alias generator, so they don't end up in query params.

5) The language parameter needs to be passed from preview controller to forwarded request.

### CODE REVIEW

**Comment by bdunogier** [comment]
About the routes, given that the whole thing is getting centralized around the concept of views, maybe we could use `/view/content` ?

**Comment by emodric** [comment]
@bdunogier That was my thought too :)

File: eZ/Bundle/EzPublishCoreBundle/Resources/config/routing/internal.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ _ezpublishLocation:
         layout: true
 
 _ez_content_view:
-    path: /content/view/{contentId}/{language}
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/Resources/config/routing/internal.yml]: As said in another comment, we could try `/view/content/{contentId}/{language}

Comment (emodric) [inline eZ/Bundle/EzPublishCoreBundle/Resources/config/routing/internal.yml]: Changed to `/view/content/{contentId}/{language}`

**Comment by bdunogier** [comment]
I'm okay with the changes here, they're good.

I'm trying to see how we can manage the `locationId` as an optional argument to this route. I'd say that it's either an extra path to the same controller (`/view/content/{contentId}/{language}/{locationId}`), or a query parameter (`/view/content/{contentId}/{language}?locationId=x`). In terms of HTTP or routing, there isn't really a difference.

What bothers me a bit more is that in the [follow-up](#1437), I only pass the request _attributes_ to the `ViewBuilder`, meaning that a query parameter wouldn't be available. Options are to a) change the `ViewBuilder` to accept the `Request` (or a clone of it), or b) give it the merge of `$request->query->all()` and `$request->attributes()->all()`.

**Comment by emodric** [comment]
Passing Request object could be a good idea to ViewBuilder::buildView. I can see that custom view builders could use any request value, not just attributes and/or query params?

**Comment by bdunogier** [comment]
> Passing Request object could be a good idea to ViewBuilder::buildView. I can see that custom view builders could use any request value, not just attributes and/or query params ?

Well, by default I prefer a looser coupling than a tighter one. For one, it keeps tests much, much more simple. For advanced usages, I'd think that you still have the option to use the Request from a custom controller to do advanced things.

But one case I can think of is potentially headers, though.There are things we could do with accept headers for instance...

**Comment by bdunogier** [comment]
We've talked about it, and we will go for route parameters. They're not as convenient in some cases, but given that the route will only be used for manual testing, I don't think it matters that much here.

Attributes can have a default value.

**Comment by bdunogier** [comment]
@lolautruche I'd like to merge this, and move on with the changes that were discussed (mostly the addition of `locationId`. +1 ?

**Comment by lolautruche** [comment]
+1

**Comment by bdunogier** [comment]
And +1 as well.

**Comment by emodric** [comment]
Thank you @bdunogier @lolautruche 

Can one of you guys do a quick merge on legacy bridge variant of this PR? https://github.com/ezsystems/LegacyBridge/pull/37

Thanks! :)

**Pull request merged**